File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    global answer
    n = len(numbers)
    answer = 0
    
    def dfs(num, Sum):
        global answer
        
        if num==n-1:
            if Sum == target:
                answer+=1
            return
        
        for i in range(2):
            numbers[num+1] = -numbers[num+1]
            Sum += numbers[num+1]
            dfs(num+1, Sum)
            Sum -= numbers[num+1]
    
    dfs(-1,0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer = 0;
    public int solution(int[] numbers, int target) {
        
        
        dfs(numbers, target, 0, 0);
        return answer;
        
    }
    public void dfs(int[] numbers, int target, int n, int sum) {
            if (n==numbers.length) {
                if (target==sum) {
                    answer++;
                }       
            } else {
                dfs(numbers, target, n+1, sum+numbers[n]);
                dfs(numbers, target, n+1, sum-numbers[n]);
            }
        }
}